build-system/tasks/extension-helpers.js

```diff
@@ -779,11 +787,15 @@ async function buildExtensionJs(dir, name, options) {
       ? wrapperOrFn(name, version, argv.esm, options.loadPriority)
       : wrapperOrFn;
 
+  // Allow the extension entry to be able to override the destination
+  // filename. This allows for a single entry point to actually have multiple
+  // output destination compilation units.
+  const destName = options.destName ?? name;
```

Comment:
Might be worth adding a type definition for `options` at this point

Reply:
will do

Reply:
looks like `ExtensionOptionDef` is already used here and is updated. I went ahead and added `CompileOptionsDef` to further add type checking in the compile/dist phase